spec/models/manageiq/providers/google/cloud_manager/provision_spec.rb

```diff
@@ -0,0 +1,35 @@
+describe ManageIQ::Providers::Google::CloudManager::Provision do
+  let(:provider) { FactoryGirl.create(:ems_google_with_authentication) }
+
+  context "Cloning" do
+    describe "#prepare_for_clone_task" do
+      let(:user_data) { "simple test user data" }
+
+      before do
+        flavor            = FactoryGirl.create(:flavor_google)
+        availability_zone = FactoryGirl.create(:availability_zone_google)
```

Comment:
Move next to `user_data`:

```ruby
let!(:flavor) { FactoryGirl.create(:flavor_google) } # Note !
# Same for availability zone
```

Otherwise great specs 😄

Reply:
I don't think you need the `!`.  Both of these will be called from the `before` block and created early.

Reply:
@bdunne That's my point, you don't need it but clean up the before block by moving all created objects to the helpers (regardless of when they need to be created) and mocking only in the before block :)

It isn't a big deal either way.

Reply:
I agree, completely 👍 for using let to get these out of the `before` block.

Reply:
@chrisarcand and @bdunne Thank you! Will do.